plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/dialogs/SQLQueryParameterBindDialog.java

```diff
@@ -183,21 +183,34 @@ public void widgetSelected(SelectionEvent e) {
                     buttonLayoutData.heightHint = editor.getSize().y;
                     button.setLayoutData(buttonLayoutData);
                     editor.selectAll();
-
+                    editor.addModifyListener(e -> saveEditorValue(editor, index, item));
                     editor.addTraverseListener(e -> {
                         if (e.detail == SWT.TRAVERSE_RETURN && (e.stateMask & SWT.CTRL) == SWT.CTRL) {
                             UIUtils.asyncExec(SQLQueryParameterBindDialog.this::okPressed);
                         }
                     });
-                    editor.addModifyListener(e -> saveEditorValue(editor, index, item));
-
+                    button.addTraverseListener(e -> {
+                        if (e.detail == SWT.TRAVERSE_TAB_NEXT) {
+                            this.keyTraversed(e);
+                        }
+                    });
                     return composite;
                 }
 
                 @Override
                 protected void saveEditorValue(Control control, int index, TableItem item) {
                     SQLQueryParameter param = (SQLQueryParameter) item.getData();
-                    String newValue = ((Text) control).getText();
+                    String newValue = null;
+                    if (!(control instanceof Text)) {
+                        Control[] children = ((Composite) control).getChildren();
+                        for (Control child : children) {
+                            if (child instanceof Text) {
+                                newValue = ((Text) child).getText();
+                            }
+                        }
+                    } else {
```

Comment:
Why is it even a case, by the way?

Reply:
When we tab out of edited cell we send composite which contains itself, button, and text. In this cases we need to find the Text child

Reply:
I don't see the place where we "send" our composite. The invocation of `keyTraversed` should not cause the issue IMO.

Reply:
Look towards the `createEditor` method.

Reply:
I don't see anything suspicious in particular in that method.

Reply:
This method is overridden inside SQLParameterBindDialog, this implementation returns not the text editor but previously mentioned composite

Reply:
Perhaps we shouldn't try to find the control of interest that way. Instead, we can bookkeep the `Text` instance as a class field and refer to it later when saving value.